src/assert.bash

```diff
@@ -308,7 +334,13 @@ refute_output() {
   fi
 
   # Matching.
-  if (( is_mode_regexp )); then
+  if (( is_mode_empty )); then
+    if [ -n "$output" ]; then
+      echo 'expected no output, but output was non-empty' \
+      | batslib_decorate 'unexpected output' \
```

Comment:
We should show `$output` here. See a few lines later in this function how to do that.